optuna/integration/allennlp.py

```diff
@@ -188,3 +224,11 @@ def __call__(
         self._trial.report(float(value), epoch)
         if self._trial.should_prune():
```

Comment:
[TODO] The current implementation re-creates a pruner (`MedianPruner`, which is a default pruning algorithm in Optuna) even if we specify a pruner in Optuna side.

Reply:
https://github.com/optuna/optuna/pull/1772#discussion_r480194645
I implemented the API to restore a pruner defined in the Optuna side.